spec/models/node_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'rails_helper'
+
+RSpec.describe Node do
```

Comment:
`reload` is used here because `n_children`, `n_descendants`, etc. are calculated by a database trigger (on `insert` only) - we need to find some way to dirty track these attributes more elegantly than asking a user to reload the model directly every time - a sure path to inconsistent behaviour and bugs.